docs/src/extend/custom-rules.md

```diff
@@ -645,6 +668,66 @@ module.exports = {
 };
 ```
 
+And as an example for the full JSON Schema object form:
```

Comment:
This sounds like the following will be the same schema, just in the JSON Schema object form, but it isn't the same. Perhaps it would be better to really show how the schema for `yoda`  rule options would look like in the JSON Schema object form?

Reply:
Sure, I was trying to show that there's more/different stuff you can do with the object schema, but I can see how it looks confusing.

I can split it into two examples, one explicitly showing the array-equivalent object schema, and a second showing off how the object schema can diverge in behaviour?